CodeEdit/Features/AppPreferences/Model/Text Editing/TextEditingPreferences.swift

```diff
@@ -27,6 +27,9 @@ extension AppPreferences {
         /// A flag indicating whether to wrap lines to editor width
         var wrapLinesToEditorWidth: Bool = true
 
+        /// The decoding that a file is read with
+        var textDecoding: TextDecodingFormats = .utf8
```

Comment:
1. Please call it `defaultTextEncoding`.

2. Use the `String.Encoding` type instead - we don't need to reimplement that.
   `String.Encoding` has the advantage of having a `rawValue` property which is of type `UInt` which will be saved, and a `description` property with a human readable presentation.

[String.Encoding Documentation](https://developer.apple.com/documentation/swift/string/encoding)

Reply:
should be fixed